utils/check_clang_tidy_results.py

```diff
@@ -9,7 +9,41 @@
 
 # Checks list: https://clang.llvm.org/extra/clang-tidy/checks/list.html
 SUPPRESSED_CHECKS = {
-    '[abseil-string-find-startswith]',
+    # Checks we don't care about and want to suppress permanently
+    '[llvm-header-guard]',  # We have our own header style with a codecheck rule on it
+    '[llvmlibc-callee-namespace]',
+    '[cert-dcl50-cpp]',  # We need this for our logger
+    '[cppcoreguidelines-pro-type-vararg]',  # We need this for our logger
+    '[hicpp-vararg]',
+    '[modernize-use-trailing-return-type]',
+    '[cppcoreguidelines-avoid-c-arrays]',
+    '[hicpp-avoid-c-arrays]',
+    '[modernize-avoid-c-arrays]',
+    '[cppcoreguidelines-avoid-goto]',
+    '[hicpp-avoid-goto]',
+    '[cppcoreguidelines-no-malloc]',
+    '[hicpp-no-malloc]',
+    '[cppcoreguidelines-non-private-member-variables-in-classes]',
+    '[misc-non-private-member-variables-in-classes]',
+    '[cppcoreguidelines-owning-memory]',
+    '[cppcoreguidelines-pro-bounds-array-to-pointer-decay]',
+    '[hicpp-no-array-decay]',
+    '[cppcoreguidelines-pro-bounds-constant-array-index]',
+    '[cppcoreguidelines-pro-bounds-pointer-arithmetic]',
+    '[cppcoreguidelines-pro-type-const-cast]',
+    '[cppcoreguidelines-pro-type-reinterpret-cast]',
+    '[cppcoreguidelines-pro-type-union-access]',
+    '[fuchsia-default-arguments]',
+    '[google-default-arguments]',
+    '[hicpp-uppercase-literal-suffix]',
+    '[readability-uppercase-literal-suffix]',
+    '[hicpp-use-auto]',
+    '[modernize-use-auto]',
+    '[modernize-raw-string-literal]',
+    '[modernize-return-braced-init-list]',
+    '[fuchsia-overloaded-operator]',
+
+    # Checks we want to clean up sometime
     '[android-cloexec-fopen]',
     '[boost-use-to-string]',
```

Comment:
Why would we ever try to clean a check that is only relevant for the usage of boost while we got completely rid of it?